cmplibrary/src/main/java/com/sourcepoint/cmplibrary/campaign/CampaignManager.kt

```diff
@@ -540,31 +542,50 @@ private class CampaignManagerImpl(
         }
 
     override val hasLocalData: Boolean
-        get() = dataStorage.gdprConsentStatus != null || dataStorage.usPrivacyString != null
-
-    override fun handleMetaDataLogic(md: MetaDataResp?) {
-        metaDataResp = md
-        md?.let {
-            it.ccpa?.apply {
-                applies?.let { i -> dataStorage.ccpaApplies = i }
-                sampleRate?.let { i ->
-                    if (i != dataStorage.ccpaSamplingValue) {
-                        dataStorage.ccpaSamplingValue = i
-                        dataStorage.ccpaSamplingResult = null
-                    }
+        get() = dataStorage.gdprConsentStatus != null || dataStorage.uspstring.isNullOrEmpty().not()
+
+    override fun handleMetaDataResponse(response: MetaDataResp?) {
+        // update meta data response in the data storage
+        metaDataResp = response
```

Comment:
@bohdan-go-wombat maybe this comment doesn't belong to this PR, but do we need to store the responses? 

If you think it makes sense, we could remove `metaDataResp` from the local storage already.

Reply:
@andresilveirah I think comments like that could actually help a lot with a readability of the code for the future refferences

regarding `metaDataResp`, I will take a look, but as I already mentioned, I strongly not recommend do a lot of refactoring in the PR, that is responsible for a speciffic fix/feature

Reply:
@bohdan-go-wombat my bad, the comment I was referring to was my GH comment, not the comment on the code 😅 

Regarding not mixing all the refactoring in the same PR, agree with you.